src/config-import.js

```diff
@@ -0,0 +1,246 @@
+      const { presets } = presetsFile
+      // sort presets using the sort field, turn them into an array
+      const sortedPresets = Object.keys(presets)
+        .filter((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          const isInvalidPreset = typeof preset !== 'object' || preset === null
+          if (isInvalidPreset) {
+            warnings.push(new Error(`invalid preset ${presetName}`))
+          }
+          return !isInvalidPreset
+        })
+        .map((presetName) => {
+          /** @type {any} */
+          const preset = presets[presetName]
+          if (!preset.sort) {
+            // if there's no sort field, put a big value - puts the field at the end -
+            preset.sort = 100
+          }
+          return preset
+        })
+        .sort((preset, nextPreset) => nextPreset.sort - preset.sort)
+
+      // 5. for each preset get the corresponding fieldId and iconId, add them to the db
+      for (let preset of sortedPresets) {
+        /** @type {any} */
+        const presetValue = {
+          schemaName: 'preset',
+          fieldIds: [],
+          addTags: {},
+          removeTags: {},
+          terms: [],
+        }
+        for (const key of Object.keys(valueSchemas.preset.properties)) {
+          if (key in preset) {
+            // @ts-ignore - we validate below
+            presetValue[key] = preset[key]
+          }
+        }
+        if (!validate('preset', presetValue)) {
+          warnings.push(new Error(`Invalid preset ${preset.name}`))
+          continue
+        }
+        yield {
+          fieldNames:
+            'fields' in preset && Array.isArray(preset.fields)
+              ? preset.fields
+              : [],
+          iconName:
+            'icon' in preset && typeof preset.icon === 'string'
+              ? preset.icon
+              : undefined,
+          value: presetValue,
+        }
+      }
+    },
+  }
+}
+
+/**
+ * @param {string} filename
+ * @param {Buffer} buf
+ * @returns {{ name: string, variant: IconData['variants'][Number] }}}
+ */
+function parseIcon(filename, buf) {
+  const matches = filename.match(
+    /^([a-zA-Z0-9-]+)-([a-zA-Z0-9]+)@(\d+x)\.(png|jpg|jpeg)$/
+  )
+  if (!matches) {
+    throw new Error(`Unexpected icon filename ${filename}`)
+  }
+  /* eslint-disable no-unused-vars */
+  const [_, name, size, pixelDensity, extension] = matches
+  const density = Number(pixelDensity.replace('x', ''))
+  if (!(density === 1 || density === 2 || density === 3)) {
+    throw new Error(`Error loading icon. invalid pixel density ${density}`)
+  }
+  if (!(size === 'small' || size === 'medium' || size === 'large')) {
+    throw new Error(`Error loading icon. invalid size ${size}`)
+  }
+  if (!name) {
+    throw new Error('Error loading icon. missing name')
+  }
+  return {
+    name,
+    variant: {
+      size,
+      mimeType: extension === 'png' ? 'image/png' : 'image/svg+xml',
+      pixelDensity: density,
```

Comment:
A few comments about this implementation:

1. I think we can simplify the `pixelDensity` capture group to avoid having to remove the `x` later, by using `(\d+)x` instead of `(\d+x)`.
1. Do we want to make the extension case-insensitive?
1. Do we want to accept SVGs too?
1. Do we want to treat JPEGs as SVGs?

```suggestion
  const parsedFilename = path.parse(filename)

  const matches = parsedFilename.name.match(/([a-zA-Z0-9-]+)-([a-z]+)@(\d)x/)
  if (!matches) {
    throw new Error(`Unexpected icon filename ${filename}`)
  }
  const [name, size, pixelDensityStr] = matches.slice(1)
  const pixelDensity = Number(pixelDensityStr)
  if (!(pixelDensity === 1 || pixelDensity === 2 || pixelDensity === 3)) {
    throw new Error(`Error loading icon. invalid pixel density ${pixelDensityStr}`)
  }
  if (!(size === 'small' || size === 'medium' || size === 'large')) {
    throw new Error(`Error loading icon. invalid size ${size}`)
  }
  if (!name) {
    throw new Error('Error loading icon. missing name')
  }

  /** @type {'image/png' | 'image/svg+xml'} */
  let mimeType
  switch (parsedFilename.ext.toLowerCase()) {
    case '.png':
      mimeType = 'image/png'
      break
    case '.svg':
      mimeType = 'image/svg+xml'
      break
    default:
      throw new Error(`Unexpected icon extension ${parsedFilename.ext}`)
  }

  return {
    name,
    variant: {
      size,
      mimeType,
      pixelDensity,
```

Reply:
1. Added that change
2. I would say yes 
3. Eventually we will support svg too, trying to match with jpeg or jpg doesn't make sense, so added your changes to allow only svg or png

Reply:
Fine to do later, but it doesn't look like the extension is case-insensitive right now.